lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -35,20 +35,20 @@ def initialize(name, instantiate=true)
     def self.globals
       hash = {}
       is_fabricpath_feature = config_get('fabricpath', 'feature')
-      return hash if (:enabled != is_fabricpath_feature.first.to_sym)
+      return hash if (:enabled != is_fabricpath_feature.to_sym)
```

Comment:
Yes, I could do that. This was leftover code when I had mixin. The feature check method was not working well with the mixin.

From: Mike Wiebe <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Friday, November 13, 2015 at 8:26 AM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] adding fabricpath topo and fixed feature_set as a class method (#75)


In lib/cisco_node_utils/fabricpath_global.rb<https://github.com/cisco/cisco-network-node-utils/pull/75#discussion_r44801697>:

> @@ -35,20 +35,20 @@ def initialize(name, instantiate=true)
>      def self.globals
>        hash = {}
>        is_fabricpath_feature = config_get('fabricpath', 'feature')
> -      return hash if (:enabled != is_fabricpath_feature.first.to_sym)
> +      return hash if (:enabled != is_fabricpath_feature.to_sym)


Just a thought. Could you rewrite the feature yaml entry to add default_value and kind features and check for enabled explicitly?

feature:
  default_value: false
  kind: boolean:
  config_get: "show feature-set"
  config_get_token: '/^fabricpath[\s]+[\d]+[\s]+(enabled)/'
  config_set: "%s feature-set fabricpath"


Then you could collapse the feature enabled check into a single line like this...

return hash unless config_get('fabricpath', 'feature')

-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/75/files#r44801697>.